src/modules/Channel/context/hooks/useStatefulFileInfoList.ts

```diff
@@ -1,30 +0,0 @@
-import { useEffect, useState } from 'react';
-import { MultipleFilesMessage, UploadableFileInfo } from '@sendbird/chat/message';
-import { createStatefulFileInfoList, StatefulFileInfo } from '../../../../utils/createStatefulFileInfoList';
-import { CoreMessageType, isMultipleFilesMessage } from '../../../../utils';
-
-export const useStatefulFileInfoList = (message: CoreMessageType): StatefulFileInfo[] => {
-  const [statefulFileInfoList, setStatefulFileInfoList] = useState<StatefulFileInfo[]>([]);
-
-  useEffect(() => {
-    if (isMultipleFilesMessage(message)) {
-      const newStatefulFileInfoList: StatefulFileInfo[] = createStatefulFileInfoList(
-        message as MultipleFilesMessage,
-        statefulFileInfoList,
-      );
-      setStatefulFileInfoList(newStatefulFileInfoList);
-    }
-  }, [
-    // Sent message dependency.
-    (message as MultipleFilesMessage).fileInfoList?.length,
-    // Unsent message dependency.
-    /**
-     * Side note: It was a bad design to not include 'isUploaded' property by SDK.
-     * Because if original object has fileUrl set and no file, then uploaded result remains
-     * the same so customer cannot know whether it has been uploaded or not.
-     */
-    (message as MultipleFilesMessage).messageParams?.fileInfoList
-      ?.map((fileInfo: UploadableFileInfo) => fileInfo.fileUrl).join(','),
-  ]);
-  return statefulFileInfoList;
-};
```

Comment:
Could you explain more what is issue with this one? 

> fix: fileInfoList should not be stateful to update (This change prevents the rendering of an empty component caused by an empty array when the mfm component is initially rendered)

Reply:
When processing a specific value derived from commonly existing data, there is no need to use state.

However, in exceptional cases where state needs to be used for such scenarios, you can set the initial value as follows:

```ts
useState(() => object.data.map(...))
```

But previously, the value was being populated in useEffect, causing a delay until the state value was filled.

```ts
useState([])

useEffect(() => {
  setState(object.data.map(...))
},[...])

```